### PR TITLE
SUS-426 In Chat switch to private room after allowing private messages

### PR DESCRIPTION
**Pull request by gitkuba**
[SUS-426](https://wikia-inc.atlassian.net/browse/SUS-426)

### CODE REVIEW

**Comment by mszabo-wikia** [comment]
LGTM

**Comment by sqreek** [comment]
LGTM

File: extensions/wikia/Chat2/js/controllers/controllers.js [hunk 1]
```diff
@@ -747,12 +747,15 @@ var NodeChatController = $.createClass(NodeRoomController, {
 			}
 
 			if (typeof(privateUser) != "undefined") {
```
Comment (WikiaTech) [inline extensions/wikia/Chat2/js/controllers/controllers.js:749]: MAJOR Replace "!=" with "!==". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 83 issues
- CRITICAL 1 critical
- MAJOR 64 major
- MINOR 4 minor
- INFO 14 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. CRITICAL [controllers.js#L101](https://github.com/Wikia/app/blob/837ca960828f000e6ef032771596bf215007a4ef/extensions/wikia/Chat2/js/controllers/controllers.js#L101): End this switch case with an unconditional break, continue, return or throw statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ANonEmptyCaseWithoutBreak)
2. MAJOR [controllers.js#L31](https://github.com/Wikia/app/blob/837ca960828f000e6ef032771596bf215007a4ef/extensions/wikia/Chat2/js/controllers/controllers.js#L31): Remove this logging statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS2228)
3. MAJOR [controllers.js#L68](https://github.com/Wikia/app/blob/837ca960828f000e6ef032771596bf215007a4ef/extensions/wikia/Chat2/js/controllers/controllers.js#L68): Remove the unused function parameter "transport". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
4. MAJOR [controllers.js#L141](https://github.com/Wikia/app/blob/837ca960828f000e6ef032771596bf215007a4ef/extensions/wikia/Chat2/js/controllers/controllers.js#L141): Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)
5. MAJOR [controllers.js#L141](https://github.com/Wikia/app/blob/837ca960828f000e6ef032771596bf215007a4ef/extensions/wikia/Chat2/js/controllers/controllers.js#L141): Replace "!=" with "!==". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)
6. MAJOR [controllers.js#L145](https://github.com/Wikia/app/blob/837ca960828f000e6ef032771596bf215007a4ef/extensions/wikia/Chat2/js/controllers/controllers.js#L145): Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)
7. MAJOR [controllers.js#L177](https://github.com/Wikia/app/blob/837ca960828f000e6ef032771596bf215007a4ef/extensions/wikia/Chat2/js/controllers/controllers.js#L177): Replace "!=" with "!==". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)
8. MAJOR [controllers.js#L217](https://github.com/Wikia/app/blob/837ca960828f000e6ef032771596bf215007a4ef/extensions/wikia/Chat2/js/controllers/controllers.js#L217): Remove the unused function parameter "index". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
9. MAJOR [controllers.js#L219](https://github.com/Wikia/app/blob/837ca960828f000e6ef032771596bf215007a4ef/extensions/wikia/Chat2/js/controllers/controllers.js#L219): Replace "==" with "===". [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AEqEqEq)
10. MAJOR [controllers.js#L232](https://github.com/Wikia/app/blob/837ca960828f000e6ef032771596bf215007a4ef/extensions/wikia/Chat2/js/controllers/controllers.js#L232): Restrict what this loop acts on by testing each property. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AForIn)

**Comment by ukaka** [comment]
After talking to @mixth-sense and @sqreek we decided to open separate ticket for fixing Sonar critical https://wikia-inc.atlassian.net/browse/SUS-946 (this was not caused by this PR). LGTM.

**Pull request merged**